### PR TITLE
Coin control in CoinJoin account

### LINKED ISSUE #6215: Coin control with anonymity level
**UTXO Categories**
1. Private - UTXOs with anonymity at least as desired anonymity level
2. Not private - UTXOs with anonymity lower then desired anonymity level
3. _Dust (already implemented)_

**UTXO Tags**
1. Registered in CoinJoin - ~~disabled from spend~~
2. Banned from CoinJoin - blacklisted or lower amount than `5000 sats`
3. _Change address (already implemented)_

User has to confirm additional check if he is using UTXOs with lower then desired anonymity level

_Note: we don't need to disable registered UTXOs from spending. Suite will not communicate registration confirmation to coordinator and UTXO will be automatically removed from CoinJoin round. Modal will prevent double-spend (CoinJoin round breakage) If we reach the critical phase_

CoinJoin - Coin control

### PR DESCRIPTION
**Pull request by komret**
## Description

The first two commits are just a minor fix and refactoring. Following commits:

- add new UTXO categories. Standard accounts still use the two categories (default + dust), while the CoinJoin account has three (private + not private + dust). Note that each category is only displayed when there are any UTXOs belonging to it. I modified behaviour of the check-all checkbox at the top so that it always selects the top category (e.g. will select all non-private UTXOs if there are no private ones. Note that dust is not recognised on Regtest as there is no `dustLimit`.
- add new UTXO tags (icons with tooltip). Apart from change address, there are now two new tags for UTXOs registered for Coinjoin and those not available for CoinJoin (too small or blocked by coordinator - the latter not yet implemented).
- add `UtxoAnonymity` component displayed in coin control and transaction detail. The same icon was used in `AnonymityLevelIndicator` to unify it. In coin control, this is displayed right from the timestamp because it has variable length.

## Related Issue

Resolve #6215

## Screenshots (if appropriate):

_Not private_ category and a UTXO with a _blocked_ tag and _1 of 1_ anonymity:
Screenshot 2022-10-31 at 17 13 02
Transaction detail showing the anonymity of my UTXO:
Screenshot 2022-10-31 at 17 13 22
New icon in `AnonymityLevelIndicator`:
Screenshot 2022-10-31 at 17 20 02

### CODE REVIEW

File: packages/suite/src/components/wallet/CoinControl/UtxoSelection.tsx [hunk 1]
```diff
@@ -123,14 +125,21 @@ interface UtxoSelectionProps {
...
     });
 
-    const { account, network, selectedUtxos, toggleUtxoSelection } = useSendFormContext();
     const theme = useTheme();
 
+    const isRegisteredForCoinjoin = coinjoinAccount?.session?.registeredUtxos.includes(
+        getUtxoOutpoint(utxo),
+    );
+    const isUnavailableForCoinjoin =
+        account.accountType === 'coinjoin' && Number(utxo.amount) < 5000; // TODO: should include UTXOs banned from CoinJoin as well
```
Comment (matejkriz) [inline packages/suite/src/components/wallet/CoinControl/UtxoSelection.tsx]: maybe BigNumber for utxo.amount? Why 5000? I don't understand this condition.

Comment (komret) [reply packages/suite/src/components/wallet/CoinControl/UtxoSelection.tsx]: Thanks, 6433f762959dbbd57722c197fdbdb73e7c013b61 explains this. It is WIP.

**Review by matejkriz — APPROVED** [review]
👍 it looks good

File: packages/suite/src/hooks/wallet/form/useUtxoSelection.ts [hunk 2]
```diff
@@ -40,19 +39,30 @@ export const useUtxoSelection = ({
...
-          )
-        : [[], []];
-
-    // are all available UTXOs selected in the form?
-    const allUtxosSelected = !!selectedUtxos.length && spendableUtxos.every(isSelected);
+    const spendableUtxos: AccountUtxo[] = [];
+    const lowAnonymityUtxos: AccountUtxo[] = [];
+    const dustUtxos: AccountUtxo[] = [];
+    account?.utxo?.forEach(utxo => {
+        switch (excludedUtxos[getUtxoOutpoint(utxo)]) {
```
Comment (dahaca) [inline packages/suite/src/hooks/wallet/form/useUtxoSelection.ts:46]: nit: perhaps rename to `excludedUtxoOutpoints?`

File: packages/suite/src/hooks/wallet/form/useUtxoSelection.ts [hunk 3]
```diff
@@ -95,7 +105,17 @@ export const useUtxoSelection = ({
         if (allUtxosSelected) {
             setValue('selectedUtxos', []);
         } else {
-            setValue('selectedUtxos', [...dustUtxos.filter(isSelected), ...spendableUtxos]);
+            setValue(
+                'selectedUtxos',
+                [
```
Comment (dahaca) [inline packages/suite/src/hooks/wallet/form/useUtxoSelection.ts]: nit: put that in a constant beforehand for readability

File: packages/suite/src/hooks/wallet/useSendForm.ts [hunk 4]
```diff
@@ -65,21 +65,17 @@ const getStateFromProps = (props: UseSendFormProps) => {
...
-        const coinjoinSession = props.coinjoinAccount?.session;
         const targetAnonymity = props.coinjoinAccount?.targetAnonymity || 1;
         const anonymitySet = account.addresses?.anonymitySet || {};
         account.utxo?.forEach(utxo => {
             const outpoint = getUtxoOutpoint(utxo);
             const anonymity = anonymitySet[utxo.address] || 1;
-            if (coinjoinSession && coinjoinSession.registeredUtxos.includes(outpoint)) {
-                // utxo is registered in coinjoin
-                excludedUtxos[outpoint] = 'mixing';
+            if (new BigNumber(utxo.amount).lte(Number(coinFees.dustLimit))) {
```
Comment (dahaca) [inline packages/suite/src/hooks/wallet/useSendForm.ts:73]: the change to `lte` is deliberate?

**Review by dahaca — COMMENTED** [review]
1. Perhaps a mention of coin control should be added to the orange warning, since it's present in the tooltip image
2. Should we have `Send not anonymized Coins` on the send button [like on the design](https://www.figma.com/file/hZ22ckyTRlCYRLTNUbTdWO/Exploring-Accounts?node-id=2195%3A36491) or keep `Review & Send`? @hynek-jina 
image

**Comment by hynek-jina** [comment]
Nice catch @dahaca 
1) Feel free to extend it to `Either anonymize more funds, lower the anonymity level, or select coins manually at your own risk`
2) I would stick to designed `Send not anonymized Coins`

File: packages/suite/src/hooks/wallet/form/useUtxoSelection.ts (hunk 2 above)
Comment (komret) [reply packages/suite/src/hooks/wallet/form/useUtxoSelection.ts:46]: I would keep this as agreed previously with @szymonlesisz. The outpoints are just pointers to UTXOs.

File: packages/suite/src/hooks/wallet/useSendForm.ts (hunk 4 above)
Comment (komret) [reply packages/suite/src/hooks/wallet/useSendForm.ts:73]: Yes, to correspond with the [validation](https://github.com/trezor/trezor-suite/blob/develop/packages/connect/src/api/signTransaction.ts#L103). It is quite unintuitive, though.

File: packages/suite/src/hooks/wallet/form/useUtxoSelection.ts (hunk 3 above)
Comment (komret) [reply packages/suite/src/hooks/wallet/form/useUtxoSelection.ts]: I refactored it so that it is more readable: daffb4d07e4e1abed98bd1d2733410a2874c55a5

**Comment by komret** [comment]
I changed the messages in c33696db7251c548814ba7309dbb04ecec077645.

Send button:
Screenshot 2022-11-01 at 19 31 57
I added a new message used when not anonymized UTXO is selected and broadcast is off:
Screenshot 2022-11-01 at 19 32 07
I removed the line break from the orange warning so that it still fits on two lines:
Screenshot 2022-11-01 at 19 34 34

I also noticed missing `margin` on `Pagination` and fixed it.

**Comment by hynek-jina** [comment]
> I changed the messages in [c33696d](https://github.com/trezor/trezor-suite/commit/c33696db7251c548814ba7309dbb04ecec077645).
> 
> Send button: Screenshot 2022-11-01 at 19 31 57 I added a new message used when not anonymized UTXO is selected and broadcast is off: Screenshot 2022-11-01 at 19 32 07 I removed the line break from the orange warning so that it still fits on two lines: Screenshot 2022-11-01 at 19 34 34
> 
> I also noticed missing `margin` on `Pagination` and fixed it.

It's a small detail.. but if that fits I would move the second sentence on the second line and add `.` dot at the end.. But I get It can be changed in Crowdin, right?

**Comment by komret** [comment]
@hynek-jina, I added the period. The sentence does not fit on one line though and it might not fit in other languages either.

**Review by dahaca — APPROVED** [review]
🙌

**Pull request merged**